File: trunk/GoServer/src/responsors/OnPacket_ReqChangeTeamIconFrame.go
package responsors

import (
	"fmt"
	//"net"
	"github.com/golang/protobuf/proto"
	. "../frame"
	. "../packs"
)

func OnPacket_ReqChangeTeamIconFrame(client *Client, pack proto.Message)() {
	//conn := client.Conn
	req := pack.(*ReqChangeTeamIconFrame)
	fmt.Println("OnPacket_ReqChangeTeamiconFrame")

	rsp := &TeamIconFrameChanged{req.Userid, req.TeamIconFrame}

	err := GetUserSystem().ChangeTeamIconFrame(req.Userid, req.TeamIconFrame);
	CheckError(err)

	
	client.Send(rsp);
}

File: trunk/Server/HallAndRoomServer/OnPacket/OnPacket_ReqChangeTeamIconFrame.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using MySql.Data.MySqlClient;


namespace HallAndRoomServer.OnPacket
{
    class OnPacket_ReqChangeTeamIconFrame
    {
        public static void OnPacket(Client client, object pack)
        {
            ReqChangeTeamIconFrame req = pack as ReqChangeTeamIconFrame;
            TeamIconFrameChanged rsp = new TeamIconFrameChanged();

            UserSystem.me.ChangeTeamIconFrame(req.Userid, req.TeamIconFrame);
            rsp.Userid = req.Userid;
            rsp.TeamIconFrame = req.TeamIconFrame;
            client.Send(rsp);            
        }
    }
}
